static/js/settings_account.js

```diff
@@ -168,7 +168,11 @@ exports.set_up = function () {
             contentType: false,
             success: function (data) {
                 loading.destroy_indicator($("#upload_avatar_spinner"));
-                $("#user-settings-avatar").expectOne().attr("src", data.avatar_url);
+                if (page_params.avatar_url_medium) {
+                    page_params.avatar_url_medium = data.avatar_url;
+                } else {
+                    page_params.avatar_url = data.avatar_url;
+                }
```

Comment:
Thanks for working on this @boxin-yang!  However, if you check out http://zulip.readthedocs.io/en/latest/new-feature-tutorial.html, you'll see that we want to make these changes inside the `server_events.js` code path, not the success handler.  For avatar changes, the `realm_user` section of `server_events.js` calls this section of `user_events.js`:

```
    if (_.has(person, 'avatar_url')) {
        var url = person.avatar_url;
        person_obj.avatar_url = url;

        if (people.is_my_user_id(person.user_id)) {
          page_params.avatar_url = url;
          $("#user-settings-avatar").attr("src", url);
        }

        message_live_update.update_avatar(person_obj.user_id, person.avatar_url);
    }
```

So probably the problem is we're not updating `avatar_url_medium` there?

Second, I think we want to set the URL to be the URL of the user's medium-size avatar, which isn't currently in the payload.  So perhaps should extend `do_change_avatar_fields` in the backend as well to send the medium-size avatar URL as well?